TripalCultivate/TripalCultivate
# Single Experiment Content Type

**Issue #25**

## Motivation
We have determined that our original approach of having multiple Experiment Type content types (i.e. Field Experiment, Greenhouse Experiment, Biochemical Assay) does not work well since many experiments may include components of two experiment types. For example, the microbiome work involves sampling from a field experiment and then sequencing to determine the species composition. Another example of this would be a food quality experiment where we need to describe the biochem assay and the way the seeds were obtained.

## What does this PR do?

As such, this PR brings together all the fields from the original Experiment Type content types into a single content type. It also strives to organize these fields in a less overwhelming way and only have one copy of any common fields (i.e. name, objectives, summary).

More specifically, this PR

1. Removes the individual experiment type content types from `tripal.tripalentitytype_collection.trpcultivate_experiments.yml` in favour of a single `research_experiment` content type.
2. Rearranges, names and combines any common fields across the 4 content types into a single content type in `tripal.tripalfield_collection.trpcultivate_experiments.yml` making sure not to loose any experiment type specific instructions.
3. Adds a default layout for the display (page view) of an experiment in `tripal_layout.tripal_layout_default_view.trpcultexperiment_layout.yml` to automate configuring the order of the many fields.
4. Adds a default layout for the form (edit) of an experiment in `tripal_layout.tripal_layout_default_form.trpcultexperiment_layout.yml` to automate configuring the order of the many fields.
5. Works to provide a consistent and intuitive organization of fields for both the experiment form and display.
6. Changes the "Germplasm Species" field to a "Germplasm Scientific Name" field to handle the situation where there are multiple genus'

## Testing

### Manual Testing

1. Create a docker on this branch.
2. Go to the content type listing and confirm we only have a single research experiment content type.
  <img width="1385" src="https://github.com/user-attachments/assets/95dfddfd-ebd4-43c9-805a-97c9a9b0d3bc">
3. Also go to the add content listing and confirm the same plus check that the definition for the research experiment vs. study is helpful, clear, accurate.
  <img width="1401" src="https://github.com/user-attachments/assets/95e15955-84e1-45e1-8cab-22a8ca599d5e">
4. Manage the form display of the Research Experiment content type, click the "Apply Tripal Layout" button and verify all the fields are present and organized in an intuitive manner.
  <img src="https://github.com/user-attachments/assets/5e3cfe43-2fc3-4f51-a7b0-ab5621d23730">
5. Manage the page display of the Research Experiment content type, click the "Apply Tripal Layout" button and verify all the fields are present and organized in an intuitive manner.
  <img src="https://github.com/user-attachments/assets/6fe3a775-3acc-493b-a985-5605e1e40dfc">
7. Create an experiment by filling out a variety of different fields and confirm it saves without error, the form is intuitive and functional, the help text is accurate and helpful, etc.
  <img src="https://github.com/user-attachments/assets/bdd88c4f-2dfa-4f3b-8bd2-d56ebe3e652b">

### Automated Testing

The automated tests are updated and just require #1993 before they will pass.
